src/store/customer/actions.js

```diff
@@ -243,3 +244,29 @@ export function createCustomer({commit, getters}) {
       commit('setIsLoading', false)
     })
 }
+
+export const applyGiftCard = ({commit}, payload) => {
+  commit('setIsLoading', true)
+  _post(
+    payload,
+    `mutation ($input: ApplyGiftCardInput) {
+      applyGiftCard(input: $input){
+        balance
+        amount
+      } 
+    }`
+  )
+    .then(({data}) => {
+      if (data.applyGiftCard !== undefined) {
+        _procAlert(data, `$${data.applyGiftCard.amount} was apllied successful`)
+        commit('setCustomerBalance', data.applyGiftCard.balance)
+      } else _procAlert(data, true)
+      commit('setIsLoading', false)
+      commit('setCurrentScannedGiftCardCode', '')
```

Comment:
commit('setIsLoading', false)
commit('setCurrentScannedGiftCardCode', '')
--> the same cleaning code could be out in finally clause